lib/syskit/shell_interface.rb

```diff
@@ -170,6 +170,45 @@ def disable_log_group(string)
         def disable_logging_of(string)
             disable_log_group(string)
         end
+
+        LoggingGroup = Struct.new(:enabled)
+        LoggingConfiguration = Struct.new(:port_logs_enabled, :conf_logs_enabled, :groups) do
+            def deep_copy
+                Marshal.load(Marshal.dump(self))
+            end
+        end
+        def logging_conf
+            conf = LoggingConfiguration.new(false, false, Hash.new)
+            conf.port_logs_enabled = Syskit.conf.logs.port_logs_enabled?
+            conf.conf_logs_enabled = Syskit.conf.logs.conf_logs_enabled?
+            Syskit.conf.logs.groups.each_pair do |key, group|
+                conf.groups[key] = LoggingGroup.new(group.enabled?)
+            end
+            conf
+        end
+        command :logging_conf, 'gets the current logging configuration'
+
+        def update_logging_conf(conf)
+            conf.port_logs_enabled ? Syskit.conf.logs.enable_port_logging :
+                                     Syskit.conf.logs.disable_port_logging
+
+            conf.conf_logs_enabled ? Syskit.conf.logs.enable_conf_logging :
+                                     Syskit.conf.logs.disable_conf_logging
+
+            conf.groups.each_pair do |name, group|
+                begin
+                    Syskit.conf.logs.group_by_name(name).enabled = group.enabled
+                rescue ArgumentError
+                    Syskit.conf.logs.create_group(name, enabled: group.enabled)
```

Comment:
This creates an empty group. Since there is no good way to create new groups for now, I would not allow creating or removing groups at all.

I would expect the following behavior:
- leave groups that are not mentioned as-is
- raise or ignore-and-warn for groups that do not exist

Reply:
Yeah, I thought about that too but then figured I could just leave the functionality there. I will remove it